### PR TITLE
[cmd/opampsupervisor,extension/opamp] Update opamp-go v0.15.0

### PR DESCRIPTION
**Pull request by BinaryFissionGames**
**Description:**
* Updates opamp-go to v0.15.0

This change is breaking, in that an opamp server using v0.14.0 may be incompatible with the changes introduced in v0.15.0 (this is due to invalid UTF-8 sequences now being allowed in the agent's instance ID).

As part of this update, the preferred format for IDs in the opamp extension's config has changed to UUID (any UUID is allowed, but specifically v7 is preferred). I've allowed ULIDs to still be specified, so older configurations should still work.

For the supervisor, I've changed the ULID generated to be a UUID. This is a breaking change for the persistent state, but this component is in developmen status, and breaking changes are expected.

**Testing:** 
Unit tests.
Tested connecting the supervisor to a management server using v0.15.0 of OpAMP.

**Documentation:**
* Modified documentation to switch references to ULID to UUID

### CODE REVIEW

File: cmd/opampsupervisor/supervisor/supervisor.go [hunk 1]
```diff
@@ -1065,26 +1065,26 @@ func (s *Supervisor) onMessage(ctx context.Context, msg *types.MessageData) {
 	}
 
 	if msg.AgentIdentification != nil {
-		newInstanceID, err := ulid.Parse(msg.AgentIdentification.NewInstanceUid)
+		newInstanceID, err := uuid.FromBytes(msg.AgentIdentification.NewInstanceUid)
 		if err != nil {
-			s.logger.Error("Failed to parse instance ULID", zap.Error(err))
-		}
+			s.logger.Error("Failed to parse instance UUID", zap.Error(err))
+		} else {
```
Comment (codeboten) [inline cmd/opampsupervisor/supervisor/supervisor.go:1071]: Is there a test for this change in behaviour?

**Review by codeboten — COMMENTED** [review]
Change looks good overall, just one question

File: cmd/opampsupervisor/supervisor/supervisor.go (hunk 1 above)
Comment (BinaryFissionGames) [reply cmd/opampsupervisor/supervisor/supervisor.go:1071]: Good catch, I added a test for this.

**Review by codeboten — APPROVED** [review]

File: cmd/opampsupervisor/supervisor/persistence_test.go [hunk 2]
```diff
@@ -48,26 +49,15 @@ func TestPersistentState_SetInstanceID(t *testing.T) {
 	require.NotEqual(t, ulid.ULID{}, state.InstanceID)
```
Comment (evan-bradley) [inline cmd/opampsupervisor/supervisor/persistence_test.go:49]:
```suggestion
	require.NotEqual(t, uuid.UUID{}, state.InstanceID)
```

File: cmd/opampsupervisor/go.mod [hunk 3]
```diff
@@ -4,12 +4,13 @@ go 1.21.0
 
 require (
 	github.com/cenkalti/backoff/v4 v4.3.0
+	github.com/google/uuid v1.6.0
 	github.com/knadh/koanf/parsers/yaml v0.1.0
 	github.com/knadh/koanf/providers/file v0.1.0
 	github.com/knadh/koanf/providers/rawbytes v0.1.0
 	github.com/knadh/koanf/v2 v2.1.1
 	github.com/oklog/ulid/v2 v2.1.0
```
Comment (evan-bradley) [inline cmd/opampsupervisor/go.mod:12]: The ulid dependency should be gone after this change, right?

File: extension/opampextension/go.mod [hunk 4]
```diff
@@ -5,7 +5,7 @@ go 1.21.0
 require (
 	github.com/google/uuid v1.6.0
 	github.com/oklog/ulid/v2 v2.1.0
```
Comment (evan-bradley) [inline extension/opampextension/go.mod:7]: Same here, we shouldn't have any remaining ulid dependencies, right?

**Review by evan-bradley — COMMENTED** [review]
Thank you for taking care of this!

File: cmd/opampsupervisor/supervisor/persistence_test.go (hunk 2 above)
Comment (BinaryFissionGames) [reply cmd/opampsupervisor/supervisor/persistence_test.go]: Good catch 👍

File: cmd/opampsupervisor/go.mod (hunk 3 above)
Comment (BinaryFissionGames) [reply cmd/opampsupervisor/go.mod]: Yeah, looks like this removed the ulid dep in the supervisor.

**Review by evan-bradley — APPROVED** [review]

**Pull request merged**